src/test/scala/firrtlTests/annotationTests/MorphismSpec.scala

```diff
@@ -0,0 +1,570 @@
+         |    inst bar of Bar
+         |  module Qux:
+         |    input x: UInt<1>
+         |    inst foo of Fub
+         |    inst bar of Bop
+         |  module Top:
+         |    inst baz of Baz
+         |    inst qux of Qux""".stripMargin
+
+    def deduped =
+      """|circuit Top:
+         |  module Foo:
+         |    node a = UInt<1>(0)
+         |    skip
+         |  module Baz:
+         |    input x: UInt<1>
+         |    inst foo of Foo
+         |    inst bar of Foo
+         |  module Top:
+         |    inst baz of Baz
+         |    inst qux of Baz""".stripMargin
+
+    def allModuleInstances =
+      IndexedSeq(
+        CircuitTarget("Top").module("Foo"),
+        CircuitTarget("Top").module("Bar"),
+        CircuitTarget("Top").module("Fub"),
+        CircuitTarget("Top").module("Bop"),
+        CircuitTarget("Top").module("Baz"),
+        CircuitTarget("Top").module("Qux"),
+        CircuitTarget("Top").module("Top")
+      )
+
+    def allAbsoluteInstances =
+      IndexedSeq(
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz").instOf("foo", "Foo"),
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz").instOf("bar", "Bar"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Qux").instOf("foo", "Fub"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Qux").instOf("bar", "Bop"),
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Qux"),
+        CircuitTarget("Top").module("Top")
+      )
+
+    def allRelative2LevelInstances =
+      IndexedSeq(
+        CircuitTarget("Top").module("Baz").instOf("foo", "Foo"),
+        CircuitTarget("Top").module("Baz").instOf("bar", "Bar"),
+        CircuitTarget("Top").module("Qux").instOf("foo", "Fub"),
+        CircuitTarget("Top").module("Qux").instOf("bar", "Bop"),
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Qux"),
+        CircuitTarget("Top").module("Top")
+      )
+
+    def allDedupedAbsoluteInstances =
+      IndexedSeq(
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz").instOf("foo", "Foo"),
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz").instOf("bar", "Foo"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Baz").instOf("foo", "Foo"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Baz").instOf("bar", "Foo"),
+        CircuitTarget("Top").module("Top").instOf("baz", "Baz"),
+        CircuitTarget("Top").module("Top").instOf("qux", "Baz"),
+        CircuitTarget("Top").module("Top")
+      )
+  }
+
+
+  behavior of "EliminateTargetPaths"
+
+  trait RightInverseEliminateTargetsFixture extends RightInverseFixture with DefaultExample {
+    override val f: Seq[Transform] = Seq(new firrtl.transforms.DedupModules)
+    override val g: Seq[Transform] = Seq(new firrtl.annotations.transforms.EliminateTargetPaths)
```

Comment:
This is a silly nit, but `EliminateTargetPaths` is actually not the inverse of `DedupModules`, due to retyping.

Reply:
Not silly, but I think it should be clear that this is an inverse where equivalence is defined structurally.